pyuvdata/uvbase.py

```diff
@@ -181,69 +203,75 @@ def __ne__(self, other):
         """
-        Check that all required parameters are set reasonably.
-
-        Check that required parameters exist and have appropriate shapes.
-        Optionally check if the values are acceptable.
+        Check that required parameters exist. Check that parameters have
+        appropriate shapes and optionally that the values are acceptable.
 
         Args:
+            check_extra: If true, check shapes and values on all parameters,
+                otherwise only check required parameters.
             run_check_acceptability: Option to check if values in required parameters
                 are acceptable. Default is True.
         """
-        for p in self.required():
+        if check_extra:
+            p_check = [p for p in self.required()] + [p for p in self.extra()]
+        else:
+            p_check = [p for p in self.required()]
+
+        for p in p_check:
             param = getattr(self, p)
             # Check required parameter exists
             if param.value is None:
-                raise ValueError('Required UVParameter ' + p +
-                                 ' has not been set.')
-
-            # Check required parameter shape
-            eshape = param.expected_shape(self)
-            if eshape is None:
-                raise ValueError('Required UVParameter ' + p +
-                                 ' expected shape is not defined.')
-            elif eshape == 'str':
-                # Check that it's a string
-                if not isinstance(param.value, str):
-                    raise ValueError('UVParameter ' + p + 'expected to be '
-                                     'string, but is not')
+                if param.required is True:
+                    raise ValueError('Required UVParameter ' + p +
+                                     ' has not been set.')
             else:
-                # Check the shape of the parameter value. Note that np.shape
-                # returns an empty tuple for single numbers. eshape should do the same.
-                if not np.shape(param.value) == eshape:
-                    raise ValueError('UVParameter {param} is not expected shape. '
-                                     'Parameter shape is {pshape}, expected shape is '
-                                     '{eshape}.'.format(param=p, pshape=np.shape(param.value),
-                                                        eshape=eshape))
-                if eshape == ():
-                    # Single element
-                    if not isinstance(param.value, param.expected_type):
-                        raise ValueError('UVParameter ' + p + ' is not the appropriate'
-                                         ' type. Is: ' + str(type(param.value)) +
-                                         '. Should be: ' + str(param.expected_type))
+                # Check parameter shape
+                eshape = param.expected_shape(self)
+                if eshape is None:
```

Comment:
I don't think this is possible. The default of `expected_shape` is to return `()`.